src/shogun/lib/DynamicArray.h

```diff
@@ -13,30 +13,875 @@
+		 * @return array element at index
+		 */
+		inline bool& element(int32_t idx1, int32_t idx2 = 0, int32_t idx3 = 0)
+		{
+			return m_array[idx1 + dim1_size * (idx2 + dim2_size * idx3)];
+		}
+
+		/** get element of given array at given index
+		 *
+		 * @param p_array another array
+		 * @param idx1 index 1
+		 * @param idx2 index 2
+		 * @param idx3 index 3
+		 * @return array element at index
+		 */
+		inline bool&
+		element(bool* p_array, int32_t idx1, int32_t idx2 = 0, int32_t idx3 = 0)
+		{
+			ASSERT(idx1 >= 0 && idx1 < dim1_size)
+			ASSERT(idx2 >= 0 && idx2 < dim2_size)
+			ASSERT(idx3 >= 0 && idx3 < dim3_size)
+			return p_array[idx1 + dim1_size * (idx2 + dim2_size * idx3)];
+		}
+
+		/** get element of given array at given index
+		 *
+		 * @param p_array another array
+		 * @param idx1 index 1
+		 * @param idx2 index 2
+		 * @param idx3 index 3
+		 * @param p_dim1_size size of dimension 1
+		 * @param p_dim2_size size of dimension 2
+		 * @return element of given array at given index
+		 */
+		inline bool& element(
+		    bool* p_array, int32_t idx1, int32_t idx2, int32_t idx3,
+		    int32_t p_dim1_size, int32_t p_dim2_size)
+		{
+			ASSERT(p_dim1_size == dim1_size)
+			ASSERT(p_dim2_size == dim2_size)
+			ASSERT(idx1 >= 0 && idx1 < p_dim1_size)
+			ASSERT(idx2 >= 0 && idx2 < p_dim2_size)
+			ASSERT(idx3 >= 0 && idx3 < dim3_size)
+			return p_array[idx1 + p_dim1_size * (idx2 + p_dim2_size * idx3)];
+		}
+
+		/** gets last array element
+		 *
+		 * @return array element at last index
+		 */
+		inline bool get_last_element() const
+		{
+			return m_array[num_elements - 1];
+		}
+
+		/** get array element at index
+		 *
+		 * (does bounds checking)
+		 *
+		 * @param index index
+		 * @return array element at index
+		 */
+		inline bool get_element_safe(int32_t index) const
+		{
+			if (index >= get_num_elements())
+			{
+				SG_SERROR(
+				    "array index out of bounds (%d >= %d)\n", index,
+				    get_num_elements());
+			}
+			return m_array[index];
+		}
+
+		/** set array element at index
+		 *
+		 * @param e element to set
+		 * @param idx1 index 1
+		 * @param idx2 index 2
+		 * @param idx3 index 2
+		 * @return if setting was successful
+		 */
+		inline bool set_element(
+		    bool e, int32_t idx1, int32_t idx2 = 0, int32_t idx3 = 0)
+		{
+			int32_t index = idx1 + dim1_size * (idx2 + dim2_size * idx3);
+			if (index < 0)
+			{
+				return false;
+			}
+			else if (index <= num_elements - 1)
+			{
+				m_array[index] = e;
+				return true;
+			}
+			else if (index < array_size)
+			{
+				m_array[index] = e;
+				num_elements = index + 1;
+				return true;
+			}
+			else
+			{
+				if (free_array && resize_array(index))
+					return set_element(e, index);
+				else
+					return false;
+			}
+		}
+
+		/** insert array element at index
+		 *
+		 * @param e element to insert
+		 * @param index index
+		 * @return if setting was successful
+		 */
+		inline bool insert_element(bool element, int32_t index)
```

Comment:
`element` shadows `this.element`:
```../src/shogun/lib/DynamicArray.h: In member function ‘bool shogun::CDynamicArray<bool>::insert_element(bool, int32_t)’:
../src/shogun/lib/DynamicArray.h:397:3: warning: declaration of ‘element’ shadows a member of 'this' [-Wshadow]``

please just use a `e` instead of `element`